src/delfino/main.py

```diff
@@ -83,17 +89,74 @@ def invoke(self, ctx: click.Context) -> Any:
     def get_help(self, *args, **kwargs) -> str:
         help_str = super().get_help(*args, **kwargs)
 
-        if self._hidden_plugins:
+        if self._hidden_commands:
             if logging.root.level == logging.DEBUG:
                 help_str += click.style(
-                    f"\n\nDisabled command{'s' if len(self._hidden_plugins) > 1 else ''}: "
-                    + ", ".join(sorted(self._hidden_plugins))
+                    f"\n\nDisabled command{'s' if len(self._hidden_commands) > 1 else ''}: "
+                    + ", ".join(sorted(self._hidden_commands))
                     + f" (see 'tool.{ENTRY_POINT}.disable_commands' in '{PYPROJECT_TOML_FILENAME}')",
                     fg="white",
                 )
 
         return help_str
 
+    def _get_commands(self) -> Dict[str, click.Command]:
+        if self._commands:
+            return self._commands
+
+        # default
+        self._commands.update(find_commands(self.PRESET_COMMAND_PACKAGE, required=True))
+
+        # external
+        self._commands.update(self.ExternalCommandsLoader(self._hidden_plugin_commands).load())
```

Comment:
> Why have you decided to use embedded class rather than a stanalone class (e.g. not nested in this one)? I think it should live in click_utils.command instead

Agree it should be in click_utils.command. Initially, I implemented a static function which returns packages. I placed the function in the Commands class as it is not related to click. When I introduced `disabled_commands` config to make the testing easier, I didn't give much though to the code location.

Reply:
Removed the class with [refactor: Remove ExternalComamndLoader](https://github.com/radeklat/delfino/pull/32/commits/e52da9d752ae8bf599b3e5d689b8ad5148de7c23)

Basically, I did the following;

- Introduce [delfino.plugin](https://github.com/radeklat/delfino/pull/32/commits/e52da9d752ae8bf599b3e5d689b8ad5148de7c23#diff-79a745e92999ad96b33c1e22edfc73e2bed5dfa3b198162af2b07161ac0029d5) module to isolate plugin load logic
- Remove ExternalComamndLoader as part of its role is now handled by `delfino.plugin.discover_plugin()`.
- Introduce _CommandRegistry to handle command registration logic which includes command name duplication handling handling
- Also added doc/comment for newly added functions and classes.

> Bonus points: This class is "a different kind of loader" to find_command. Can you come up with a common interface for both, so they can be simply run in a loop? Some things might go into __init__, some into __call__ (which is what load() is at the moment).

@radeklat I couldn't identity good interface could you elaborate on this more?